synapse/handlers/register.py

```diff
@@ -55,6 +56,16 @@
     ["guest", "auth_provider"],
 )
 
+LoginDict = TypedDict(
+    "LoginDict",
+    {
+        "device_id": str,
+        "access_token": str,
+        "valid_until_ms": Optional[int],
+        "refresh_token": Optional[str],
+    },
+)
```

Comment:
I did not manage to make that work in a worker setup because it can go through the [`RegisterDeviceReplicationServlet`](https://github.com/matrix-org/synapse/pull/9450/files#diff-a3bbf601c798f027761fde58f9e58bb5a50dab2bd5017597747760f3975b1cc6L77) if it is in a worker setup, and that has to return dict I think?

Reply:
fair enough.